### ISSUE TITLE
Focus not updated when HA is in "Allowed Notifications" list

### ISSUE THREAD

**Issue by davidolrik** [home-assistant/iOS#2094]
I've seen this issue on my own device and in a few issues already reported (#1899, #2057).

It seems to me like a bug that the `binary_sensor.focus` is not updated when Home Assistant is added to the "Allowed Notifications" list for a given focus.

Focus should *always* be updated when given the permission to do so, even if HA is in the "Allowed Notifications" list.

As it is now, if I want notifications to go through when a focus is active, I have to mark them as `time-sensitive`.
But doing this means they will go through in **all** focus modes I have - which is not what I want, as not everything is `time-sensitive`.

**Comment by zacwest** [comment]
Unfortunately from the app's perspective, focus status is not set when the app is in that list. There's no way to detect that you're in this state or that focus is actually set. This is how Apple designed the system to function.

Open to ideas on how to make this more obvious. Where did you go in the app to try and figure out what was going wrong? Did you open the Sensors preferences and tap into Focus at all?

**Comment by davidolrik** [comment]
I had created an elaborate way of getting the name of the focus pushed into a `input_text` helper, which stopped working at some point.
I then tried debugging it using a `entities` card with everything focus related. Googling led me to #2057, which made me re-read the docs - which doesn't make sense, because it seems counterintuitive that being in the "Allowed notifications" list should prevent the focus state from being exposed.

I don't know how to make it more obvious, because it is totally counterintuitive.
